documents/MIGRATION_GUIDE.md

```diff
@@ -0,0 +1,1243 @@
+> The [get_stats](https://awslabs.github.io/aws-crt-python/api/mqtt5.html#awscrt.mqtt5.Client.get_stats) method returns
+> the current state of an `awscrt.mqtt5.Client` object's queue of operations in
+> [OperationStatisticsData](https://awslabs.github.io/aws-crt-python/api/mqtt5.html#awscrt.mqtt5.OperationStatisticsData),
+> which may help with tracking the number of in-flight messages.
+
+``` python
+op_stats_data = mqtt5_client.get_stats()
+print(
+    "Client operations queue statistics:\n" +
+    "incomplete_operation_count:" + op_stats_data.incomplete_operation_count() + "\n"
+    "incomplete_operation_size: " + op_stats_data.incomplete_operation_size() + "\n"
+    "unacked_operation_count: " + op_stats_data.unacked_operation_count() + "\n"
+    "unacked_operation_size: " + op_stats_data.unacked_operation_size() + "\n")
+
+```
+
+For more information, see [withOfflineQueueBehavior documentation](https://awslabs.github.io/aws-crt-python/api/mqtt5.html#awscrt.mqtt5.ClientOptions).
+
+For the list of the supported offline queue behaviors and their descriptions,
+see [ClientOfflineQueueBehavior types documentation](https://awslabs.github.io/aws-crt-python/api/mqtt5.html#awscrt.mqtt5.ClientOperationQueueBehaviorType).
+
+### Operation Timeouts
+
+In the v1 SDK, all operations (*publish*, *subscribe*, *unsubscribe*) will not timeout unless
+you define a timeout for them. If no timeout is defined, there is a possibility that an operation will wait forever
+for the server to respond and block the calling thread indefinitely.
+
+In the v2 SDK, operations timeout is set for the MQTT5 client with the
+[ClientOptions](https://awslabs.github.io/aws-crt-python/api/mqtt5.html#awscrt.mqtt5.ClientOptions)
+class member `ack_timeout_sec`.
+The default value is no timeout. As in the v1 SDK, failing to set a timeout can cause an operation to stuck forever,
+but it won't block the client.
+
+The [get\_stats](https://awslabs.github.io/aws-crt-python/api/mqtt5.html#awscrt.mqtt5.Client.get_stats) method returns
+the current state of an` mqtt5.Client` object's queue of operations, which may help with tracking operations.
+
+#### Example of timeouts in the v1 SDK
+
+```python
+connectTimeoutSec = 10
+`client``.``configureConnectDisconnectTimeout`(connectTimeoutSec)
+client.connect();
+
+ publishTimeoutMs = 20
+client.`configureMQTTOperationTimeout``(publishTimeoutMs)`
+client.publish("my/topic", "hello", 1)
+
+```
+
+### Example of timeouts in the v2 SDK
+
+```python
+mqtt5_client = mqtt5_client_builder.mtls_from_path(
+        endpoint="<prefix>-ats.iot.<region>.amazonaws.com",
+        cert_filepath="<certificate file path>",
+        pri_key_filepath="<private key file path>",
+        ack_timeout_sec=20)
+
+```
+
+### Logging
+
+The v1 SDK uses `AWSIoTPythonSDK.core` custom logger logger for logging.
+
+The v2 SDK uses a standard  [logging facility](https://docs.python.org/3/howto/logging.html).
```

Comment:
In the example below we use a logging from crt:
```
from awscrt import io
io.init_logging(log_level=io.LogLevel.Trace, file_name='stderr')
```

Reply:
What I mean is the v2 sdk uses the logger provided by crt, not the standard one.